rust/src/builtins/compose/mod.rs

```diff
@@ -6,13 +6,29 @@ pub(crate) mod commit;
 
 use crate::cxxrsutil::CxxResult;
 use anyhow::{Context, Result};
+use cap_std_ext::rustix;
 use fn_error_context::context;
 use openat_ext::OpenatDirExt;
 use std::ffi::{CStr, CString};
 use std::io::{self, Read};
 
 use crate::core::OSTREE_BOOTED;
 
+/// Display information about the target filesystem.
+pub fn composeutil_print_target_info(rootfs_dfd: i32) -> CxxResult<()> {
+    let rootfs_dfd = unsafe { rustix::fd::BorrowedFd::borrow_raw_fd(rootfs_dfd) };
+    let statfs = rustix::fs::fstatfs(&rootfs_dfd)?;
+    // Or https://github.com/uutils/coreutils/blob/64f3cd748d5a188d5123790110d1d8ca7b33f18d/src/uucore/src/lib/features/fsext.rs#L748
+    println!(
+        "Target filesystem metadata:
+  Type: {} (Go to <https://github.com/coreutils/coreutils/blob/1118f32145049cc4b949e89eda89a0ed612755de/src/stat.c#L219> to reverse this)
```

Comment:
Maybe 

```suggestion
  Type: {:#x} (Go to <https://github.com/coreutils/coreutils/blob/1118f32145049cc4b949e89eda89a0ed612755de/src/stat.c#L219> to reverse this)
```

so it prints in hex to make it easier to compare?

Reply:
Hmm, really tempting to just copy in that `pretty_fstype` function. WDYT?

Reply:
Yeah, OK done